src/engraving/editing/cmd.cpp

```diff
@@ -3375,6 +3374,108 @@ void Score::cmdIncDecDuration(int nSteps, bool stepDotted)
+
+void Score::cmdExtendToNextNote()
+{
+    const Fraction startTick = selection().tickStart();
+    Fraction endTick = selection().tickEnd();
+    const staff_idx_t startStaff = selection().staffStart();
+    const staff_idx_t endStaff = selection().staffEnd();
+
+    std::vector<ChordRest*> toSelect;
+    const bool wasRangeSelection = selection().isRange();
+
+    for (EngravingItem* el : selection().elements()) {
+        if (!el->isNote()) {
+            continue;
+        }
+        Note* n = toNote(el);
+        ChordRest* cr = toChordRest(n->chord());
+        toSelect.push_back(cr);
+        InputState& is = score()->inputState();
+
+        if (cr->endTick() == lastSegment()->tick()) {
+            continue;
+        }
+
+        // fill next empty measure with rest for voices > 0
+        if (cr->voice() != 0 && !nextChordRest(cr)) {
+            Measure* m = cr->measure()->nextMeasure();
+            is.setTrack(cr->track());
+            is.setSegment(m->segments().firstCRSegment());
+            score()->enterRest(m->ticks());
+        }
+
+        ChordRest* ncr = nextChordRest(cr);
+
+        while (ncr->isRest() || (ncr->tick() > cr->endTick())) { //second condition checks for empty measure between cr and ncr for voices 2,3,4.
+            if (cr->endTick() != ncr->tick()) { // check for empty measures and fill with mRest for voices>0
+                is.setTrack(cr->track());
+                is.setSegment(cr->measure()->nextMeasure()->segments().firstCRSegment());
+                score()->enterRest(cr->measure()->nextMeasure()->ticks());
+                ncr = nextChordRest(cr);
+            }
+            if (cr->tuplet() || ncr->tuplet()) {
+                m_is.setSegment(ncr->segment());
+                m_is.setLastSegment(m_is.segment());
+                m_is.setDuration(ncr->durationType());
+
+                Note* nn = nullptr;
+                for (size_t i = 0; i < toChord(cr)->notes().size(); i++) {
+                    Note* note = toChord(cr)->notes()[i];
+                    NoteVal nval(note->noteVal());
+                    nn = addPitch(nval, i != 0);
+
+                    Tie* tie =  Factory::createTie(note);
+                    tie->setStartNote(note);
+                    tie->setTrack(note->track());
+                    tie->setTick(note->chord()->segment()->tick());
+                    tie->setEndNote(nn);
+                    tie->setTicks(nn->chord()->segment()->tick() - note->chord()->segment()->tick());
+                    undoAddElement(tie);
+                }
+                cr = toChordRest(nn->chord());
+                toSelect.push_back(cr);
+            } else {
+                Fraction newDur = cr->ticks() + ncr->ticks();
+                changeCRlen(cr, newDur);
+                Fraction tickAtDurEnd = cr->tick() + newDur;
+                while (nextChordRest(cr) && tickAtDurEnd > nextChordRest(cr)->tick()) { // if resulting cr has ties
+                    cr = nextChordRest(cr);
```

Comment:
You could add an auxiliary variable here to avoid calling `nextChordRest(cr)` thrice with the same `cr`